core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -130,7 +130,13 @@ private[deploy] class Master(
     logInfo(s"Running Spark version ${org.apache.spark.SPARK_VERSION}")
     webUi = new MasterWebUI(this, webUiPort)
     webUi.bind()
-    masterWebUiUrl = "http://" + masterPublicAddress + ":" + webUi.boundPort
+    val SSL_ENABLED = conf.getBoolean("spark.ssl.enabled", false)
+    val uriScheme = "http://"
+    if (SSL_ENABLED) {
+      uriScheme = "https://"
```

Comment:
have you tested this? This is a val, I doubt this can even compile...

Reply:
Could you try this? It's a bit closer to the usual scala style.

```
    val uriScheme = if (conf.getBoolean("spark.ssl.enabled", false)) "https://" else "http://"
```

Reply:
woops, i just updated val to var

Reply:
yes, but still please add tests for this, in order to enforce the correctness of your solution. PS I agree with @pjfanning's suggestion..